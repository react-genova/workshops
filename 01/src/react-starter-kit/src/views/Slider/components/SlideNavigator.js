//@flow
import React from 'react';
import styled from 'styled-components';

const Container = styled.div`
    position: absolute;
    z-index: 9999;
    left: 0;
    bottom: 0;
    top: 0;
    right: 0;
`;

const NavBar = styled.div`
    display: flex;
    align-items: center;
    flex-direction: row;
    justify-content: center;
    width: 3em;
    height: 100%;
    background-color: transparent;
    opacity: 0;
    cursor: pointer;
    &:hover {
        opacity: 1;
    }
`;

const Next = NavBar.extend`
    float: right;
`;

const Previous = NavBar.extend`
    float: left;
`;

const Arrow = styled.div`
    width: 0; 
    height: 0; 
    border-top: 2em solid transparent;
    border-bottom: 2em solid transparent;
`;
const ArrowNext = Arrow.extend`
    border-left: 2em solid #264d73;
`;

const ArrowPrev = Arrow.extend`
    border-right: 2em solid #264d73;
`;

type Props = {
    onNext:()=>void,
    onPrev:()=>void,
};

const SlideNavigator = ({onNext, onPrev}:Props) => (
    <Container>
        <Next onClick={onNext}><ArrowNext /></Next>
        <Previous onClick={onPrev}><ArrowPrev /></Previous>
    </Container>
);

export default SlideNavigator;