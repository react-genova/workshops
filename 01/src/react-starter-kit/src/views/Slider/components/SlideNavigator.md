```js
<div style={{width:'400px', height:'150px', backgroundColor:'#edf1f8', position:'relative'}}>
    <SlideNavigator onNext={()=>alert('Next clicked')} onPrev={()=>alert('Previous clicked')} />
</div>
```