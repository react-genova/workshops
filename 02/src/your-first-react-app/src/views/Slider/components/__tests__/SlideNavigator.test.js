//@flow
import React from 'react';
import SlideNavigator from '../SlideNavigator';
import { shallow } from 'enzyme';

describe('SlideNavigator component', () => {

    it('simple header snapshot', () => {
        const fake = () => {};
        const component = shallow(<SlideNavigator onNext={fake} onPrev={fake} />);
        expect(component).toMatchSnapshot();
    });

});