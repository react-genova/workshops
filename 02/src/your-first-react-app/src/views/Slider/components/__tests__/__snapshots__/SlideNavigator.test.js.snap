// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SlideNavigator component simple header snapshot 1`] = `
<styled.div>
  <styled.div
    onClick={[Function]}
  >
    <styled.div />
  </styled.div>
  <styled.div
    onClick={[Function]}
  >
    <styled.div />
  </styled.div>
</styled.div>
`;
